File: tests/R/test_trilinear.R
# Test for gridfit trilinear_rcpp dummy output
library(gridfit)

x <- c(0, 1)
y <- c(0, 1)
z <- c(0, 1)
values <- c(1, 2, 3, 4, 5, 6, 7, 8)
points <- matrix(c(0.5, 0.5, 0.5), ncol=3, byrow=TRUE)
result <- trilinear_rcpp(x, y, z, values, points)
stopifnot(all(result == 4.5))

File: tests/python/test_trilinear.py
# Test for gridfit.trilinear dummy output
import numpy as np
from gridfit import GridFit


def test_trilinear_dummy():
    x = np.array([0.0, 1.0], dtype=np.float32)
    y = np.array([0.0, 1.0], dtype=np.float32)
    z = np.array([0.0, 1.0], dtype=np.float32)
    values = np.array(
        [
            [[1, 2], [3, 4]],
            [[5, 6], [7, 8]],
        ],
        dtype=np.float32,
    )
    points = np.array([[0.5, 0.5, 0.5]], dtype=np.float32)
    gridfit = GridFit(x, y, z, values)
    out = gridfit.interpolate(points)
    print(out)
    assert out[0] == 4.5
